deploy/cdk/marble-elasticsearch-pipeline/marble_elasticsearch_pipeline/marble_elasticsearch_pipeline_stack.py

```diff
@@ -0,0 +1,190 @@
+        super().__init__(scope, id, **kwargs)
+        self.es_stack = 'marble-elasticsearch'
+        self.repo_name = repo_name
+        self.repo_branch = repo_branch
+        self.repo_owner = repo_owner
+        self._setup_artifact()
+        self._setup_iam()
+        self.pipeline = codepipeline.Pipeline(self, 'ElasticSearchPipeline',
+                                              artifact_bucket=self.artifact_bucket,
+                                              role=self.codepipeline_role)
+
+    def add_stages(self):
+        # REPOSITORY STAGE
+        # Need a cloudformation parameter to keep the token from being exposed in the template
+        oauth_desc = 'Secret. OAuthToken with access to Repo. Long string of characters and digits. Go to https://github.com/settings/tokens'
+        oauth_token_param = core.CfnParameter(self, 'GitHubToken',
+                                              type='String', no_echo=True,
+                                              min_length=8,
+                                              description=oauth_desc)
+        source_output = codepipeline.Artifact()
+        source_action = codepipeline_actions.GitHubSourceAction(action_name='Github_App_Source',
+                                                                owner=self.repo_owner,
+                                                                repo=self.repo_name,
+                                                                branch=self.repo_branch,
+                                                                oauth_token=core.SecretValue.cfn_parameter(oauth_token_param),
+                                                                output=source_output,
+                                                                trigger=codepipeline_actions.GitHubTrigger.POLL)
+        self.pipeline.add_stage(stage_name='Source', actions=[source_action])
+
+        # TEST STAGE
+        test_project = build.BuildProject(self, self.codebuild_role, 'test', {'es_stackname': self.es_stack}).pipeline_project()
+        test_actions = codepipeline_actions.CodeBuildAction(action_name=f"testBuildMarbleElasticSearchStack",
+                                                            project=test_project,
+                                                            input=source_output,
+                                                            run_order=1)
+
+        # SNS topics
+        approval_topic = sns.Topic(self, 'PipelineApprovalTopic', display_name='PipelineApprovalTopic')
+        approval_actions = codepipeline_actions.ManualApprovalAction(action_name='ManualApprovalOfTestEnvironment',
+                                                                     notification_topic=approval_topic,
+                                                                     additional_information='Approve or Reject this change after testing',
```

Comment:
Idk if you can do more in this case, but try to make sure to leave an informative message here. This text is what populates the slack approval message. If there's no message, it will just have a link to the pipeline that triggered the approval message. I usually try to point to test/prod urls if available, or at least the stack names.